controllers/infra/controllers/infra_controller.go

```diff
@@ -82,16 +91,36 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		// requeue after 30 seconds
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, err
 	}
-	if res == controllerutil.OperationResultUpdated {
-		if infra.Status.Status == infrav1.Running.String() {
-			return ctrl.Result{}, nil
-		}
-		infra.Status.Status = infrav1.Running.String()
+	// clean infra using aws terminate
+	if !infra.DeletionTimestamp.IsZero() {
```

Comment:
Encapsulate it, this piece is too complicated. Refer to the implementation of user controller